packages/xo-server/src/api/test.js

```diff
@@ -86,3 +89,31 @@ copyVm.resolve = {
   vm: ['vm', 'VM'],
   sr: ['sr', 'SR'],
 }
+
+// -------------------------------------------------------------------
+
+export async function xapiPoolChange({
+  currentPoolObject,
+  newPoolHostname,
+  newPoolObject,
+}) {
+  const xapi = this.getXapi(currentPoolObject)
+  const { pool: currentPool } = xapi
+
+  xapi._setUrl({ ...xapi._url, hostname: newPoolHostname })
+  await fromEvent(xapi.objects, 'finish')
+  if (xapi.pool.$id === currentPool.$id) {
+    await fromEvent(xapi.objects, 'finish')
+  }
+
+  assert(xapi.pool.$id !== currentPool.$id)
+  assert.doesNotThrow(() => this.getXapi(newPoolObject))
+  assert.throws(() => this.getXapi(currentPoolObject))
+}
+
+xapiPoolChange.permission = 'admin'
+xapiPoolChange.params = {
+  currentPoolObject: { type: 'string' },
+  newPoolHostname: { type: 'string' },
+  newPoolObject: { type: 'string' },
+}
```

Comment:
I'm not against merging this but we have to improve clarity:

- rename to `changeConnectedXapiHostname`
- add description explaining the purpose of this method and what it does
- params: `hostname`, `oldObject` and `newObject`
- add description on the params